src/core/lib/surface/call.cc

```diff
@@ -1445,6 +1445,13 @@ static grpc_call_error call_start_batch(grpc_call* call, const grpc_op* ops,
           call->send_initial_metadata.Set(grpc_core::GrpcTimeoutMetadata(),
                                           call->send_deadline);
         }
+
+        // On the server side, grpc-timeout metadata should not
+        // be passed.
+        if (!call->is_client) {
```

Comment:
We could probably omit this `if` check and move the code up a little - on the client we want to be in charge of it to.

Reply:
Done.